app.py

```diff
@@ -1,38 +1,110 @@
-from flask import Flask, render_template, request, redirect, url_for
+from flask import Flask, render_template, request, redirect, url_for, flash
 import mongo as mongo
 import view_model as view_model
 import dotenv
+from flask_login import login_required, login_user, current_user
+import login_manager as login_manager
+from oauthlib.oauth2 import WebApplicationClient
+import os
+import requests
+import json
+from user import User
+from flask_config import Config
 
 def create_app():
     app = Flask(__name__)
+    app.config.from_object(Config())
+    app.config['LOGIN_DISABLED'] = os.environ.get('LOAD_DISABLED', 'False').lower() in ['true', '1']
+    login_manager.login_manager.init_app(app)
 
     @app.route('/', methods=['Get'])
+    @login_required
     def index():
         items = mongo.fetch_all_items()
         items.sort(key=lambda k: k.status, reverse=True)
         item_view_model = view_model.ViewModel(items)
-        return render_template('index.html', view_model=item_view_model) 
+        if current_user.is_active == True:
+            if current_user.role == 'writer':
+                return render_template('index_writer.html', view_model=item_view_model)
+            elif current_user.role == 'reader':
+                return render_template('index_reader.html', view_model=item_view_model)
+            else:
+                return render_template('index_writer.html', view_model=item_view_model)
+        else:
+            return render_template('index_writer.html', view_model=item_view_model)
```

Comment:
is_active is True if the user is "active"... i.e. you can have users with disabled accounts -> is_active is False.

You do not want to treat inactive users as Writers.

I think you meant to check LOGIN_DISABLED rather than whether current_user.is_active